ycms/static/src/js/ward/discharge.ts

```diff
@@ -0,0 +1,103 @@
+import { Modal, ModalOptions } from "flowbite";
+
+let selectedBedId: string | null = "";
+let selectedBedAssignmentId: string | null = "";
+
+export const handleModalClose = (): void => {
+    document.getElementById("cancelDischargeButton")?.addEventListener("click", () => {
+        const modal = (FlowbiteInstances as any).getInstance("Modal", "modalEl");
+        modal.hide();
+    });
+    document.getElementById("closeModalButton")?.addEventListener("click", () => {
+        const modal = (FlowbiteInstances as any).getInstance("Modal", "modalEl");
+        modal.hide();
+    });
+};
+
+export const handleConfirmDischarge = (): void => {
+    document.getElementById("confirmDischargeButton")?.addEventListener("click", () => {
+        const getCookie = (name: string): string | null => {
+            let cookieValue = null;
+            if (document.cookie && document.cookie !== "") {
+                const cookies = document.cookie.split(";");
+                for (let i = 0; i < cookies.length; i++) {
+                    const cookie = cookies[i].trim();
+                    // Check if the cookie name matches the provided name
+                    if (cookie.substring(0, name.length + 1) === `${name}=`) {
+                        cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+                        break;
+                    }
+                }
+            }
+            return cookieValue;
+        };
+
+        // Use AJAX to mark the patient as discharged
+        fetch(`discharge-patient/${selectedBedAssignmentId}/${selectedBedId}/`, {
+            method: "POST", // Use POST or PATCH for partial updates
+            headers: {
+                "Content-Type": "application/json",
+                "X-CSRFToken": getCookie("csrftoken") || "", // Include CSRF token
+            },
+            body: JSON.stringify({ discharged: true }),
+        })
+            .then((response) => response.json())
+            .then((data) => {
+                if (data.success) {
+                    // If the POST request was successful, refresh the page
+                    handleModalClose();
+                    // eslint-disable-next-line no-restricted-globals
+                    location.reload();
+                } else {
+                    // Handle errors if needed
+                    console.error("POST request failed");
+                }
```

Comment:
```suggestion
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "X-CSRFToken": getCookie("csrftoken") || "",
            },
        })
            .then((response) => response.json())
            .then((data) => {
                if (data.success) {
                    handleModalClose();               
                    window.location.reload();
                } else {
                    console.error("POST request failed");
                }
```
I think the comments were pretty self explanatory :)

The `body` data is not necessary, it's just what you get back from the view, right?